nexus/src/app/update/mod.rs

```diff
@@ -56,17 +56,30 @@ impl super::Nexus {
             ArtifactsWithPlan::from_stream(body, Some(file_name), &self.log)
                 .await
                 .map_err(|error| error.to_http_error())?;
-
-        // Now store the artifacts in the database.
         let tuf_repo_description = TufRepoDescription::from_external(
             artifacts_with_plan.description().clone(),
         );
-
+        // Move the `ArtifactsWithPlan`, which carries with it the
+        // `Utf8TempDir`s storing the artifacts, into the artifact replication
+        // background task. This is done before the database insert because if
+        // this fails, we shouldn't record the artifacts in the database.
+        self.tuf_artifact_replication_tx
+            .send(artifacts_with_plan)
+            .await
+            .map_err(|err| {
+                Error::internal_error(&format!(
+                    "failed to send artifacts for replication: {err}"
+                ))
+            })?;
+        // Now store the artifacts in the database.
```

Comment:
If we send `artifacts_with_plan`, write to the database, and crash before the background task completes, what happens to the state durably recorded in the database? Wouldn't we have a `tuf_repo` recorded in a database table, that doesn't actualyl exist on the rack?

Reply:
We would. There's a `// TODO` in the background task to mark individual repositories or artifacts as "failed" if any of the artifacts are verifiably missing. But this is not yet implemented.

However I think given the comment below I'm going to flip this logic. We still need to implement some way of noticing missing artifacts though.